pandas/tests/io/test_excel.py

```diff
@@ -361,6 +361,35 @@ def test_reader_dtype(self, ext):
         with pytest.raises(ValueError):
             actual = self.get_exceldf(basename, ext, dtype={'d': 'int64'})
 
+    def test_reader_dtype_str(self, ext):
+        # GH 20377
+        basename = 'testdtype'
+        actual = self.get_exceldf(basename, ext)
+
+        expected = DataFrame({
+            'a': [1, 2, 3, 4],
+            'b': [2.5, 3.5, 4.5, 5.5],
+            'c': [1, 2, 3, 4],
+            'd': [1.0, 2.0, np.nan, 4.0]}).reindex(
+                columns=['a', 'b', 'c', 'd'])
+
+        tm.assert_frame_equal(actual, expected)
+
+        actual = self.get_exceldf(basename, ext,
+                                  dtype={'a': 'float64',
+                                         'b': 'float32',
+                                         'c': str,
+                                         'd': str})
+
+        expected['a'] = expected['a'].astype('float64')
+        expected['b'] = expected['b'].astype('float32')
+        expected['c'] = ['001', '002', '003', '004']
+        expected['d'] = ['1', '2', '', '4']
+        tm.assert_frame_equal(actual, expected)
+
+        with pytest.raises(ValueError):
```

Comment:
What's the error message?

Reply:
```
ValueError: Unable to convert column d to type int64
```
which is the same as in the `test_reader_dtype`, which I copied it from. No reason to have this here I guess.

Reply:
> No reason to have this here I guess.

What do you mean?